corehq/apps/app_manager/util.py

```diff
@@ -409,46 +409,83 @@ def _add_ancestors(ctype, generation):
     else:
         generations = []
 
+    subsets = [{
+        "id": generation_names[i],
+        "name": "{} ({})".format(generation_names[i], " or ".join(ctypes)) if i > 0 else base_case_type,
+        "key": "@case_type",
+        "structure": {p: {} for type in [map[t] for t in ctypes] for p in type},
+        "related": {"parent": {
+            "hashtag": "#case/" + generation_names[i + 1],
+            "subset": generation_names[i + 1],
+            "key": "@case_id",
+        }} if i < len(generations) - 1 else None,
+    } for i, ctypes in enumerate(generations)]
+
+    if is_usercase_in_use(app.domain):
```

Comment:
If we want this rebranding to happen sooner rather than later, I think this is good to merge if you add a feature flag to enable user case in the form builder. That way QA can look at this on prod and this PR won't grow and grow.

Would also be nice to be able to get rid of the legacy structure out of vellum